README.md

```diff
@@ -911,140 +771,210 @@ final class PostStory extends Story
+This library requires that your database be reset before each test. The packaged `ResetDatabase` trait handles this for
+you. Before the first test, it drops (if exists) and creates the test database. Before each test, it resets the schema.
 
-        // create 50 Post's
-        PostFactory::new()->createMany(50, function() {
-            return [
-                // each Post will have a random Category (created above)
-                'category' => CategoryFactory::random(),
+```php
+use Zenstruck\Foundry\Test\Factories;
+use Zenstruck\Foundry\Test\ResetDatabase;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
 
-                // each Post will between 0 and 6 Tag's (created above)
-                'tags' => TagFactory::randomRange(0, 6),
-            ];
-        });
-    }
+class MyTest extends WebTestCase // TestCase must be an instance of KernelTestCase
+{
+    use ResetDatabase, Factories;
+    
+    // ...
 }
 ```
 
-Run the [`doctrine:fixtures:load`](https://symfony.com/doc/master/bundles/DoctrineFixturesBundle/index.html#loading-fixtures)
-as normal to seed your database.
+**TIP**: Create a base TestCase for tests using factories to avoid adding the traits to every TestCase.
 
-### Global Test State
+By default, `ResetDatabase` resets the default configured connection's database and default configured object manager's
+schema. To customize the connection's and object manager's to be reset (or reset multiple connections/managers), set the
+following environment variables:
 
-If you have an initial database state you want for all tests, you can set this in your `tests/bootstrap.php`:
+```.env.test
+FOUNDRY_RESET_CONNECTIONS=connection1,connection2
+FOUNDRY_RESET_OBJECT_MANAGERS=manager1,manager2
+```
+
+### Auto-Refresh & Force Setting
+
+Objects created by a factory are wrapped in a special [`Proxy`](src/Proxy.php) object. These objects help
+with your *pre and post act* test assertions. Almost all calls to Proxy methods, first refresh the object from the
+database (even if your entity manager has been cleared).
 
 ```php
-// tests/bootstrap.php
-// ...
+use App\Factory\PostFactory;
+use Zenstruck\Foundry\Proxy;
 
-Zenstruck\Foundry\Test\TestState::addGlobalState(function () {
-    CategoryFactory::create(['name' => 'php']);
-    CategoryFactory::create(['name' => 'symfony']);
+$post = PostFactory::new()->create(['title' => 'My Title']); // instance of Zenstruck\Foundry\Proxy
+
+// get the wrapped object
+$post->object(); // instance of Post
+
+// delete from the database
+$post->remove();
+
+// call any Post methods - before calling, the object is auto-refreshed from the database
+$post->getTitle(); // "My Title"
+
+// set property and save to the database
+$post->setTitle('New Title');
+$post->save(); 
+
+/**
+ * CAVEAT - When calling multiple methods that change the object state, the previous state will be lost because
+ * of auto-refreshing. Use "disableAutoRefresh()" or "withoutAutoRefresh()" to overcome this.
+ */
```

Comment:
```php
$post->getTitle(); // orig title
$post->getBody(); // orig body
$post->setTitle('new title');
$post->setBody('new body');
$post->save();
$post->getTitle(); // orig title (because the subsequent call to `setBody()` auto-refreshed the object and overrode the unpersisted title)
$post->getBody(); // new body
```

Is the problem, this is the solution:

```php
$post->getTitle(); // orig title
$post->getBody(); // orig body
$post->disableAutoRefresh(); // !!
$post->setTitle('new title');
$post->setBody('new body');
$post->save(); // saves and re-enables auto-refreshing
$post->getTitle(); // new title
$post->getBody(); // new body
```

Having a hard time getting that across...

Reply:
https://github.com/zenstruck/foundry/issues/18